server.go

```diff
@@ -1048,6 +1050,13 @@ func (s *server) AnnounceNewTransactions(newTxs []*mempool.TxDesc) {
 	}
 }
 
+// Transaction has one confirmation on the main chain. Now we can mark it as no
+// longer needing rebroadcasting.
+func (s *server) TransactionConfirmed(tx *btcutil.Tx) {
+	iv := wire.NewInvVect(wire.InvTypeTx, tx.Hash())
```

Comment:
This doesn't need to be done unless the RPC server is active because the only time rebroadcasting is needed is for transactions that were submitted via RPC. That check seems to have been lost in the shuffle.

I think it's quite reasonable to have this call separate to decouple that logic, but really should have the check for a nil RPC server (aka no RPC server is running) moved into it.

The following should do the trick nicely:


```Go
// No need to work with rebroadcasting when the RPC server is not active.
if s.rpcServer == nil {
	return
}
```